### PR TITLE
FEInterface::map() did fail for infinite elements

### PR DESCRIPTION
**Pull request by BalticPinguin**
I figured out that FEInterface::map() does not work for infinite elements. Supposing this is not a feature, I added the respective functions. Therefore, I needed to change InfFE::map() from protected to public, but I don't see problems with it. Are there?

Best,
Hubert

### CODE REVIEW

File: src/fe/inf_fe_map.C [hunk 1]
```diff
@@ -486,17 +486,18 @@ Point InfFE<Dim,T_radial,T_map>::inverse_map (const Elem * inf_elem,
...
-          //  }
+           if (p(2)+1 < tolerance)
+            {
+           libmesh_here();
+           libMesh::err << "WARNING: radial distance p(2) is "
+                << p(2)
+                << std::endl;
+            }
+          */
 #endif
```
Comment (roystgnr) [inline src/fe/inf_fe_map.C]: Why the commenting changes?

**Comment by roystgnr** [comment]
I'd like more eyes on it before we merge, but looks reasonable to me, thanks.

**Comment by BalticPinguin** [comment]
I uncommented it for my personal use and than inserted them in the wrong way, so you can ignore them, sorry for that.

By the way: If you want, I can write an extra report on that but I get there errors when inverse_mapping and mapping again of up to  1.13449 (or 0.2/0.3 ) at some points. For me this seems to be an unreasonable large error, isn't it?

File: src/fe/inf_fe_map.C (hunk 1 above)
Comment (jwpeterson) [inline src/fe/inf_fe_map.C]: Please either undo the commenting change or simply delete the code altogether.  There's no reason for this to be part of the diff...

File: src/fe/inf_fe_map.C [hunk 2]
```diff
@@ -592,18 +593,16 @@ Point InfFE<Dim,T_radial,T_map>::inverse_map (const Elem * inf_elem,
...
-    }
-  */
+    //if (diff.size() > tolerance)
+    //{
+    //   libmesh_here();
+    //   libMesh::err << "WARNING:  diff is "
+    //   << diff.size()
+    //   << std::endl;
+    //}
 #endif
```
Comment (jwpeterson) [inline src/fe/inf_fe_map.C]: Same goes for this commenting change.

File: src/fe/fe_interface_inf_fe.C [hunk 3]
```diff
@@ -425,6 +425,30 @@ void FEInterface::ifem_nodal_soln(const unsigned int dim,
 
 
 
+Point FEInterface::ifem_map (const unsigned int dim, 
+                             const FEType& fe_t,
+                             const Elem * elem,
+                             const Point & p){
+   switch (fe_t.inf_map){
```
Comment (jwpeterson) [inline src/fe/fe_interface_inf_fe.C]: Curly braces in this entire function need to go on their own line to match our coding conventions.

File: src/fe/fe_interface.C [hunk 4]
```diff
@@ -539,6 +539,11 @@ Point FEInterface::map(unsigned int dim,
                        const Elem * elem,
                        const Point & p)
 {
+#ifdef LIBMESH_ENABLE_INFINITE_ELEMENTS
+
+  if (is_InfFE_elem(elem->type()))
```
Comment (jwpeterson) [inline src/fe/fe_interface.C]: I hadn't noticed this before, but this function's name does not match our usual naming scheme... that should probably be fixed, but not in this PR.

File: include/fe/fe_interface.h [hunk 5]
```diff
@@ -394,6 +394,11 @@ class FEInterface
                               const std::vector<Number> & elem_soln,
                               std::vector<Number> & nodal_soln);
 
+  static Point ifem_map (const unsigned int dim, 
+                                      const FEType& fe_t,
```
Comment (jwpeterson) [inline include/fe/fe_interface.h]: Indentation is not correctly aligned for the new function.

**Comment by jwpeterson** [comment]
> when inverse_mapping and mapping again of up to 1.13449 (or 0.2/0.3 ) at some points. For me this seems to be an unreasonable large error, isn't it?

Yes, this sounds like a bug.  Can you get this result prior to making your changes, or is it something introduced by this patch?  A minimal working example posted as a gist would be helpful, but I'm not sure any of us would really be able to debug the math any time soon...

**Comment by BalticPinguin** [comment]
Ok, I will do the respective changes in the patch. 
The error in the mapping seems should not come from my code since I don't touch the calculations itself; I will investigate it more and than write a report/patch for it separately.

**Comment by jwpeterson** [comment]
> Ok, I will do the respective changes in the patch.

Thanks.  Just do `git commit --amend` after making the formatting changes, and then `git push -f` to update the PR.



> The error in the mapping seems should not come from my code since I don't touch the calculations itself; I will investigate it more and than write a report/patch for it separately.

OK, it's possible that it is a long-standing bug.  If you can get it resolved, we should also add some basic unit testing for the infinite elements code path...

**Comment by BalticPinguin** [comment]
sorry for the many commits; I missed the format of the comments.

**Comment by jwpeterson** [comment]
> sorry for the many commits; I missed the format of the comments.

No problem, just use [interactive rebase](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) to squash them down into a single commit. 

Also, don't forget to fix the opening curly braces in fe_interface_inf_fe.C.  They should all be on a separate line.

**Comment by BalticPinguin** [comment]
nice! Thanks for that hint.
Now it looks fine, if I didn't miss something.

**Comment by jwpeterson** [comment]
Thanks for making the formatting changes.  I fixed a couple others and merged, but keep us up to date about the issues you were having with `ifem_map`.

**Pull request merged**